tests/AbstractFixerTestCase.php

```diff
@@ -0,0 +1,658 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Paysera\PhpCsFixerConfig\Tests;
+
+use Exception;
+use InvalidArgumentException;
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\AbstractProxyFixer;
+use PhpCsFixer\Fixer\Comment\HeaderCommentFixer;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Fixer\DeprecatedFixerInterface;
+use PhpCsFixer\Fixer\FixerInterface;
+use PhpCsFixer\Fixer\Whitespace\SingleBlankLineAtEofFixer;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionInterface;
+use PhpCsFixer\FixerDefinition\CodeSampleInterface;
+use PhpCsFixer\FixerDefinition\FileSpecificCodeSampleInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSampleInterface;
+use PhpCsFixer\Linter\CachingLinter;
+use PhpCsFixer\Linter\Linter;
+use PhpCsFixer\Linter\LinterInterface;
+use PhpCsFixer\Linter\ProcessLinter;
+use PhpCsFixer\Preg;
+use PhpCsFixer\StdinFileInfo;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\WhitespacesFixerConfig;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use ReflectionMethod;
+use SplFileInfo;
+use PhpCsFixer\Tokenizer\Token;
+
+use function array_slice;
+use function count;
+use function in_array;
+use function is_string;
+use function sprintf;
+
+use const PHP_VERSION_ID;
+
+abstract class AbstractFixerTestCase extends TestCase
+{
+    /**
+     * @var FixerInterface|ConfigurableFixerInterface|null
+     */
+    protected ?FixerInterface $fixer = null;
+    protected ?LinterInterface $linter = null;
+
+    private array $allowedRequiredOptions = [
+        'header_comment' => ['header' => true],
+    ];
+
+    // do not modify this structure without prior discussion
+    private array $allowedFixersWithoutDefaultCodeSample = [
+        'general_phpdoc_annotation_remove' => true,
+        'general_phpdoc_tag_rename' => true,
+    ];
+
+    abstract protected function createFixer();
+
+    final public function testIsRisky()
+    {
+        if ($this->fixer->isRisky()) {
+            self::assertValidDescription($this->fixer->getName(), 'risky description', $this->fixer->getDefinition()->getRiskyDescription());
+        } else {
+            self::assertNull($this->fixer->getDefinition()->getRiskyDescription(), \sprintf('[%s] Fixer is not risky so no description of it expected.', $this->fixer->getName()));
+        }
+
+        if ($this->fixer instanceof AbstractProxyFixer) {
+            return;
+        }
+
+        $reflection = new ReflectionMethod($this->fixer, 'isRisky');
+
+        // If fixer is not risky then the method `isRisky` from `AbstractFixer` must be used
+        self::assertSame(
+            !$this->fixer->isRisky(),
+            AbstractFixer::class === $reflection->getDeclaringClass()->getName()
+        );
+    }
+
+    final public function testFixerDefinitions()
```

Comment:
Refactoring is needed, method length is almost 180+ lines...

I guess this is a result of not following https://github.com/paysera/php-style-guide?tab=readme-ov-file#composition-over-inheritance

Reply:
Yes, refactoring is needed. This method was taken from the old php-cs-fixer library, from AbstractFixerTestCase. Let's fix it after the merge.